nixos/modules/services/networking/juicity.nix

```
@@ -0,0 +1,89 @@
+{ pkgs
+, config
+, lib
+, ...
+}:
+with lib;
+let
+  cfg = config.services.juicity;
+in
+{
+  meta = {
+    maintainers = with lib.maintainers; [ oluceps ];
+  };
+
+  options.services.juicity = {
+    instances = mkOption {
+      description = "list of juicity instance";
+      type = types.listOf (types.submodule {
+        options = {
+          name = mkOption {
+            type = types.str;
+            description = "juicity instance name";
+          };
+          package = mkPackageOption pkgs "juicity" { };
+          credentials = mkOption {
+            type = types.listOf types.str;
+            default = [ ];
+            description = ''
+              Load extra credentials.
+              Could be written as systemd `LoadCredentials` format e.g.
+              `["key:/etc/juicity-key"]` and access in config with
+              `/run/credentials/juicity-$\{name}.service/key`
+            '';
+          };
+          openFirewall = mkOption {
+            type = with types; nullOr port;
+            default = null;
+            description = "Open firewall port";
+          };
+          serve = mkEnableOption "Use `juicity-server` instead of `juicity-client` if set `true`";
```

```suggestion
          serve = mkEnableOption "using `juicity-server` instead of `juicity-client`";
```

`mkEnableOption` is "Whether to enable <...>."